WooCommerce/Classes/Notifications/PushNotificationsManager.swift

```diff
@@ -213,6 +234,46 @@ extension PushNotificationsManager {
     }
 }
 
+// MARK: - Notification count & app badge number update
+//
+private extension PushNotificationsManager {
+    func incrementNotificationCount(siteID: Int64, type: Note.Kind, incrementCount: Int, onCompletion: @escaping () -> Void) {
+        let action = NotificationCountAction.incrementNotificationCount(siteID: siteID, type: type, incrementCount: incrementCount, onCompletion: onCompletion)
+        ServiceLocator.stores.dispatch(action)
+    }
+
+    func loadNotificationCountAndUpdateApplicationBadgeNumberAndPostNotifications(siteID: Int64, type: Note.Kind?) {
+        loadNotificationCountAndUpdateApplicationBadgeNumber(siteID: siteID)
+        postBadgeReloadNotifications(type: type)
+    }
+
+    func loadNotificationCountAndUpdateApplicationBadgeNumber(siteID: Int64) {
+        let action = NotificationCountAction.loadNotificationCount(siteID: siteID, type: nil) { [weak self] count in
+            self?.configuration.application.applicationIconBadgeNumber = count
+        }
+        ServiceLocator.stores.dispatch(action)
```

Comment:
⚠️  Same as above, we don't need to use `ServiceLocator` since there is an injected `configuration`.

Reply:
similar to previous comments, updated in https://github.com/woocommerce/woocommerce-ios/pull/2320/commits/7620bb0da4f901ebfd141939e5511a9ab657c39e